cpp/src/gandiva/llvm_types.h

```diff
@@ -57,14 +58,20 @@ class GANDIVA_EXPORT LLVMTypes {
 
   llvm::PointerType* ptr_type(llvm::Type* type) { return type->getPointerTo(); }
 
+  llvm::PointerType* opaque_ptr_type() { return ptr_type(i8_type()); }
```

Comment:
shouldn't opaque_ptr_type be ptr_type(void_type)?; any reason why it is ptr_type() of i8_type

Reply:
That's because LLVM doesn't support opaque pointer types yet, see http://nondot.org/~sabre/LLVMNotes/EliminatingVoid.txt and https://llvm.org/docs/OpaquePointers.html.

Will add a comment here, though.

Reply:
We've recently added support for opaque pointers